File: app/src/main/java/com/abhi/lokaldemo/di/ActivityBuilder.java
package com.abhi.lokaldemo.di;

import com.abhi.lokaldemo.ui.main.MainActivity;
import com.abhi.lokaldemo.ui.main.MainActivityModule;
import com.abhi.lokaldemo.ui.main.MainFragment;

import dagger.Module;
import dagger.android.ContributesAndroidInjector;

@Module
public abstract class ActivityBuilder {

    @ContributesAndroidInjector(modules = {MainActivityModule.class})
    abstract MainActivity bindMainActivity();

}

File: app/src/main/java/com/abhi/lokaldemo/di/ActivityBuilder.kt
package com.abhi.lokaldemo.di

import dagger.Module
import com.abhi.lokaldemo.ui.main.MainActivity
import com.abhi.lokaldemo.ui.main.MainActivityModule
import dagger.android.ContributesAndroidInjector



@Module
abstract class ActivityBuilder {


    @ContributesAndroidInjector(modules = [MainActivityModule::class])
    internal abstract fun bindMainActivity(): MainActivity
}